### PR TITLE
Upgrade to OpenSSL3

### PR DESCRIPTION
**Pull request by bkabrda**
* Add an OpenSSL 3 omnibus software config
* For other configs that depend on openssl, allow choosing whether to depend on openssl 1 or openssl 3 through an environment variable
* Remove the openssl dependency from several recipes that don't actually need it (bzip2, nghttp2)
* Upgrade several other recipes to oldest versions that work with openssl 3 (curl, postgres)

### CODE REVIEW

File: config/software/openssl3.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  # Out of abundance of caution, we put the feature flags first and then
+  # the crazy platform specific compiler flags at the end.
+  configure_args << env["CFLAGS"] << env["LDFLAGS"]
+
+  configure_command = configure_args.unshift(configure_cmd).join(" ")
+
+  command configure_command, env: env, in_msys_bash: true
+
+  patch source: "openssl-3.1.0-do-not-build-docs.patch", env: env
```
Comment (KSerrania) [inline config/software/openssl3.rb:88]: Why is the patch prefixed with `openssl-3.1.0` if we're building 3.0.9?

Comment (bkabrda) [reply config/software/openssl3.rb:88]: I started the work with OpenSSL 3.1.0, but then I found out that Python 3.9 isn't compatible with that, so I downgraded OpenSSL to 3.0.9. I can certainly rename it to 3.0.9 if you think that's better, but it does cleanly apply to both versions.

Comment (KSerrania) [reply config/software/openssl3.rb:88]: 👍, let's rename for clarity.

**Review by KSerrania — APPROVED** [review]
LGTM.

**Pull request merged**